system/mp/mpcomm.cpp

```diff
@@ -831,13 +832,23 @@ protected: friend class CMPPacketReader;
 #ifdef _TRACE
                                 EXCLOG(e, "MP: Failed to connect");
 #endif
+                                // if other side closes because of stale connection on same host:port, connect again
+                                if (e->errorCode() == JSOCKERR_graceful_close)
+                                {
+                                    LOG(MCdebugInfo(100), unknownJob, "MP: Retrying (other side closed connection, probably due to clash)");
+                                    e->Release();
+                                    break;
```

Comment:
I will put this into a new JIRA issue.  I can look at where it should go and take retrycount/timeout into account.